File: TP-2/money_js/src/money_ops.js
var MoneyOps=function (){
}

MoneyOps.add = function(m1,m2){
	if(m1.getCurrency().toUpperCase() !== m2.getCurrency().toUpperCase())
		throw new DevisesIncompatibleExc(m1.getCurrency().toUpperCase() , m2.getCurrency().toUpperCase());

	return new money(m1.getValue()+m2.getValue(),m1.getCurrency());
};

MoneyOps.sub = function(m1,m2){
    if(m1.getCurrency().toUpperCase() !== m2.getCurrency().toUpperCase())
        throw new DevisesIncompatibleExc(m1.getCurrency().toUpperCase() , m2.getCurrency().toUpperCase());

    if(m1.getValue() < m2.getValue())
        throw new SoustractionValueExc(m1.getValue() , m2.getValue());

    return new money(m1.getValue()-m2.getValue(),m1.getCurrency());
};

File: TP-1/money_src/src/fr/ulille1/fil/odeva/MoneyOps.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package fr.ulille1.fil.odeva;

/**
 *
 * @author marius
 */
public class MoneyOps {
  private static MoneyFactory mf = MoneyFactory.getDefaultFactory();

  /**
   * Add moneys having the same currency
  */
  public static Money simpleAdd(Money m1, Money m2) throws UnexistingCurrencyException {
    if(!m1.getCurrency().equals(m2.getCurrency()))
      throw new IncompatibleCurrencyException(m1.getCurrency(),m2.getCurrency());

    Money m=mf.createMoney(m1.getValue()+m2.getValue(),m1.getCurrency());
    return m;
  }

  /**
   * Add moneys having the dif currency
   */
  public static Money simpleSub(Money m1, Money m2) throws IncompatibleCurrencyException, ValueException, UnexistingCurrencyException {
    if(!m1.getCurrency().equals(m2.getCurrency()))
      throw new IncompatibleCurrencyException(m1.getCurrency(),m2.getCurrency());

    if(m1.getValue() < (m2.getValue()))
      throw new ValueException(m1.getValue(),m2.getValue());

    Money m=mf.createMoney(m1.getValue()+m2.getValue(),m1.getCurrency());
    return m;
  }
}
